app/controllers/invitations_controller.rb

```diff
@@ -0,0 +1,25 @@
+class InvitationsController < ApplicationController
+  before_filter :require_user
+
+  def new
+    @invitation = Invitation.new
+  end
+
+  def create
+    @invitation = Invitation.new(invitation_params.merge!(inviter_id: current_user.id))
+    if @invitation.save
+      AppMailer.send_invitation_email(@invitation).deliver_now
+      flash[:success] = "You have successfully invited #{@invitation.recipient_name}."
+      redirect_to new_invitation_path
+    else
+      flash[:danger] = "Please check your inputs."
```

Comment:
When you render its always `flash.now` otherwise the message sticks to the next request as well.